client/settings/general-settings-section/payment-methods-list.js

```diff
@@ -160,6 +160,27 @@ const StyledFees = styled( PaymentMethodFeesPill )`
 	flex: 1 0 auto;
 `;
 
+/**
+ * Formats the payment method description with the account default currency.
+ *
+ * @param {*} method Payment method ID.
+ * @param {*} accountDefaultCurrency Account default currency.
+ */
+const getFormattedPaymentMethodDescription = (
+	method,
+	accountDefaultCurrency
+) => {
+	const { description, acceptsDomesticPaymentsOnly } = PaymentMethodsMap[
+		method
+	];
+
+	if ( acceptsDomesticPaymentsOnly ) {
+		return sprintf( description, accountDefaultCurrency.toUpperCase() );
```

Comment:
```suggestion
		return sprintf( description, accountDefaultCurrency?.toUpperCase() );
```

Need a safety check here. I was getting the following error in console with a blank Stripe settings page.

```
Uncaught TypeError: Cannot read properties of undefined (reading 'toUpperCase')
    at getFormattedPaymentMethodDescription
```

Reply:
To be more specific, I had the API keys of a US based account and store currency set to Euro/CAD when I got this error.

<img src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/5b59f362-7865-4241-838d-f5fbe3d60ac0" width="50%" height="50%" />

Reply:
Thanks for the suggestion @Mayisha. I wasn't able to replicate this but happy to be more defensive here. I think you'd need to have an account without a default currency set on the account at all. 🤷‍♂️